### PR TITLE
Add additional redirects for archives

### PR DESCRIPTION
**Pull request by thaJeztah**
Putting these after the existing redirects, so that ee-specific redirects in the archives take precedence.

### CODE REVIEW

**Comment by thaJeztah** [comment]
@usha-mandya @StefanScherer PTAL

**Comment by thaJeztah** [comment]
We'll have to test if they're added in the right position; I assumed that it processes redirects in order (so if there's an existing `/v18.09/ee/` redirect higher up the list, it would be processed first, and other `/v18.09/` will be processed by the later rule).

File: _website-config-docs.json [hunk 1]
```diff
@@ -344,6 +344,45 @@
...
+    },
+    {
+      "Condition": {
+        "HttpErrorCodeReturnedEquals": null,
+        "KeyPrefixEquals": "v18.09/"
+      },
+      "Redirect": {
+        "HostName": "docs.docker.com",
+        "HttpRedirectCode": null,
+        "Protocol": "https",
```
Comment (thaJeztah) [inline _website-config-docs.json:382]: I saw in the docs that these are optional, so thinking of doing a follow-up to test if redirects work ok without these specified (in that case, we could possibly use a single config both for `docs-stage` and for `docs`

Comment (StefanScherer) [reply _website-config-docs.json:382]: I think I tried that, but maybe I did it wrong.
Anyhow the feedback loop is 
1. create PR
2. merge PR to master
3. test redirects on docs-stage.docker.com
4. revert PR if it broke something

Comment (StefanScherer) [reply _website-config-docs.json:382]: The other idea I had before adding two json files was, we could specify the HostName value when running the GitHub action https://github.com/docker/docker.github.io/blob/master/.github/actions/update-website-config/main.go that uploads the website config in the workflow https://github.com/docker/docker.github.io/blob/master/.github/workflows/build-master.yml#L27-L32

Comment (StefanScherer) [reply _website-config-docs.json:382]: I think I remember, when you leave that HostName away, then the internal URL to the S3 bucket is used after the redirect without the docker.com domain.

Comment (thaJeztah) [reply _website-config-docs.json:382]: Ah, that'd make sense yes, that's a pity

**Comment by thaJeztah** [comment]
Tested this on staging and looks to work ok;

https://docs-stage.docker.com/v18.09/engine/install/ redirects to https://docs-stage.docker.com/engine/install/
and https://docs-stage.docker.com/v18.09/ee/ still redirects to the mirantis website

**Comment by usha-mandya** [comment]
Redirect rules are live in prod. Tested this in prod and it all works fine.

**Pull request merged**